src/ui/components/pages/NewWalletName.tsx

```diff
@@ -1,174 +1,148 @@
+        setError(t('newAccountName.errorRequired'));
+      }
 
-    if (names.includes(name)) {
-      errors.push({
-        code: 2,
-        key: 'newAccountName.errorInUse',
-        msg: 'Name already exist',
-      });
+      if (accounts.find(({ name }) => name === accountName)) {
+        setError(t('newAccountName.errorInUse'));
+      }
     }
 
-    return errors;
-  }
-
-  static getDerivedStateFromProps(props, state) {
-    const { account, accounts } = props;
-
-    return {
-      ...state,
-      errors: NewWalletNameComponent.validateName(
-        (account && account.name) || '',
-        accounts
-      ),
-    };
-  }
-
-  onChange = e => this._onChange(e);
-
-  onSubmit = e => this._onSubmit(e);
-
-  onBlur = () => this._onBlur();
-
-  render() {
-    return (
-      <div className={styles.content}>
-        <h2 className={`title1 margin1`}>
-          <Trans i18nKey="newAccountName.accountName">Account name</Trans>
-        </h2>
-
-        <form onSubmit={this.onSubmit}>
-          <div className={`margin1`}>
-            <Input
-              id="newAccountName"
-              className="margin1"
-              onChange={this.onChange}
-              value={this.props.account.name || ''}
-              maxLength="32"
-              autoFocus={true}
-              error={this.state.error}
-              onBlur={this.onBlur}
-            />
-            <Error show={this.state.error} errors={this.state.errors} />
+    if (existedAccount) {
+      setError(
+        t('newAccountName.errorAlreadyExists', {
+          name: existedAccount.name,
+        })
+      );
+    }
+  }, [accountName, accounts]);
+
+  return (
+    <div data-testid="newWalletNameForm" className={styles.content}>
+      <h2 className={`title1 margin1`}>
+        <Trans i18nKey="newAccountName.accountName" />
+      </h2>
+
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+
+          setPending(true);
+
+          if (existedAccount) {
+            dispatch(selectAccount(existedAccount));
+            return setTab(PAGES.IMPORT_SUCCESS);
+          }
+
+          if (error) {
+            return;
+          }
+
+          const accountTypeToWalletType = {
+            seed: WalletTypes.Seed,
+            encodedSeed: WalletTypes.EncodedSeed,
+            privateKey: WalletTypes.PrivateKey,
+            wx: WalletTypes.Wx,
+            ledger: WalletTypes.Ledger,
+          };
+
+          dispatch(
+            createAccount(account, accountTypeToWalletType[account.type])
+          );
+        }}
+      >
+        <div className={`margin1`}>
+          <Input
+            data-testid="newAccountNameInput"
+            className="margin1"
+            onChange={e => setAccountName(e.target.value)}
+            value={accountName}
+            maxLength="32"
+            disabled={existedAccount}
+            autoFocus
+            error={error}
+            onFocus={() => setTouched(true)}
```

Comment:
Was it changed from `onBlur` to `onFocus` intentionally? Because of `autoFocus`, `onFocus` is called right away. So in that case it doesn't make sense to think about `touched` state at all, it will be `false` only for short period of time after mount.